### ISSUE TITLE
Checkout after clone fails on Windows 10 due to filename length error

### ISSUE THREAD

**Issue by asampal** [enso-org/enso#1266]
Just tried to clone and got the output below:

```
[10:08 PM] Clone: cannot create directory at 'lib/scala/runtime-version-manager-test/src/main/resources/org/enso/runtimeversionmanager/test/fake-releases/enso/enso-0.999.0-marked-broken/enso-engine-0.999.0-marked-broken-linux-amd64.tar.gz/enso-0.999.0-marked-broken': Filename too long
	c:\Temp\smartgit15142107960096193691> git.exe -c credential.helper=D:/dev/SmartGit/lib/credentials.cmd clone -v --progress --branch main https://github.com/enso-org/enso.git D:\dev\projects-open-source\enso
	Cloning into 'D:\dev\projects-open-source\enso'...
	POST git-upload-pack (725 bytes)
	remote: Total 23705 (delta 77), reused 330 (delta 56), pack-reused 23263        
	cannot create directory at 'lib/scala/runtime-version-manager-test/src/main/resources/org/enso/runtimeversionmanager/test/fake-releases/enso/enso-0.999.0-marked-broken/enso-engine-0.999.0-marked-broken-linux-amd64.tar.gz/enso-0.999.0-marked-broken': Filename too long
	Clone succeeded, but checkout failed.
	You can inspect what was checked out with 'git status'
	and retry with 'git restore --source=HEAD :/'
```

**Comment by radeusgd** [comment]
I can indeed reproduce the issue on Windows.

@asampal A very simple and quick fix will be to try cloning the repository into a location with a 'shorter name', for example directly to `D:\dev\enso` (I keep it in a similar place and it works this way) or even `D:\enso`. Of course this is not ideal, but it is the quickest solution to get it working for now.

It seems that this length limit has been [removed in Windows update 1607, but disabling the limit is opt-in](https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation). So another workaround could be to just disable the limit as described in the documentation (although I do not know and cannot guarantee if and how it may affect other applications).

@iamrecursion What do you think we should do about it in longer-term?
I suggest we may want to add into the contributing doc an information that due to path length limits the repo should be checked out in a 'shallow' directory and/or link to the documentation on disabling the limit in Windows 10.
Alternatively, we could change the way the test fixtures are set up for the release tests (as I think these are the only problematic files). However the latter will only 'postpone' the issue as if contributors check out the repo in a deeper directory the problem will come back.

**Comment by 4e6** [comment]
This issue can be fixed in git config (v1.9.0 required):

```
git config core.longpaths true
```

The comment on the issue PR warns that the above setting may break some scripts, but I was able to successfully build and run the project.

```
https://github.com/msysgit/git/pull/122#issuecomment-36782101
```
